examples/preprocessing/plot_callable_transformer.py

```diff
@@ -0,0 +1,69 @@
+"""
+=========================================================
+Using CallableTransformer to select columns
+=========================================================
+
+Shows how to use a callable transformer in a pipeline. If you know your
+dataset's first principle component is irrelevant for a classification task,
+you can use the CallableTransformer to select all but the first column of the
+PCA transformed data.
+"""
+import matplotlib.pyplot as plt
+import numpy as np
+
+from sklearn.cross_validation import train_test_split
+from sklearn.decomposition import PCA
+from sklearn.pipeline import make_pipeline
+from sklearn.preprocessing import CallableTransformer
+
+
+def _generate_vector(shift=0.5, noise=15):
+    return np.arange(1000) + (np.random.rand(1000) - shift) * noise
+
+
+def generate_dataset():
+    """
+    This dataset is two lines with a slope ~ 1, where one has
+    a y offset of ~100
+    """
+    return np.vstack((
+        np.vstack((
+            _generate_vector(),
+            _generate_vector() + 100,
+        )).T,
+        np.vstack((
+            _generate_vector(),
+            _generate_vector(),
+        )).T,
+    )), np.hstack((np.zeros(1000), np.ones(1000)))
+
+
+def all_but_first_column(X, y):
+    return X[:, 1:]
+
+
+def drop_first_component(X, y):
```

Comment:
This just factors the work being done in to named components so that users can read through the code more easily. I don't mind changing it but I am not convinced that a single linear script is more readable.